panacea/web/static/js/view.js

```diff
@@ -139,10 +156,47 @@ export const restoreFileForm = () => {
   hideElement(loadingElement);
 };
 
+const generateErrorHighlight = (lines, errorLine, numContextLines = 2) => {
+  const errorlineIndex = errorLine - 1;
+
+  const startIndex = Math.max(0, errorlineIndex - numContextLines);
+  const endIndex = Math.min(lines.length-1, errorlineIndex + numContextLines);
+  const displayLines = lines.slice(startIndex, endIndex + 1); // slice extracts up to but not including end
+
+  const codeBlockElement = createElementWithClass('PRE','code-block');
+
+  displayLines.forEach((lineContents, currentIndex) => {
```

Comment:
Not too familiar with JS idioms, but in other places we're using:

```javascript
for (const x of xs)
```

Whereas here we use `.forEach(...)`. Any particular reason? If not, I'd go for consistency everywhere.